dynamic_programming/abbreviation.cpp

```diff
@@ -0,0 +1,164 @@
+/**
+ * @file
+ * @brief Implementation of Abbrievation
+ * (https://www.hackerrank.com/challenges/abbr/problem)
+ *
+ * @details
+ * Given two strings, a and b, determine if it's possible to make a equal to
+ * b You can perform the following operations on the string a:
+ * 1. Capitalize zero or more of a's lowercase letters.
+ * 2. Delete all of the remaining lowercase letters in a.
+ *
+ * ##Algorithm
+ * The idea is in the problem statement itself: iterate through characters of
+ * string a and b (for character indexes i and j respectively):
+ * 1. If a[i] and b[j] are equal, then move to next position
+ * 2. If a[i] is lowercase of b[j], then explore two possibilities:
+ *  a. Capitalize or
+ *  a. Skip a[i]
+ * 3. If the a[i] is not uppercase, just discard the char, else return false
+ *
+ * Time Complexity (O(|a|*|b|)) where |a| => length of string
+ * @author Ashish Daulatabad (https://github.com/AshishYUO)
+ */
+
+#include <cassert>
+#include <iostream>
+#include <string>
+#include <vector>
```

Comment:
Add a brief description of what the header/library is for (see the example below).
```c++
#include <cassert>     /// for assert
#include <iostream>    /// for IO operations
```